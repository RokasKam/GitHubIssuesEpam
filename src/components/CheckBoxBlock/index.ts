import { CheckBoxBlock } from "./CheckBoxBlock";
export { CheckBoxBlock };