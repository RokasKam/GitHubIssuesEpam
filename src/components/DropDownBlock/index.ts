import { DropDownBlock } from "./DropDownBlock";
export { DropDownBlock };