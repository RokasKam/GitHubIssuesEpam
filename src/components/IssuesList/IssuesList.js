import React from 'react'
import { FlatList } from 'react-native';
import { styles } from './IssuesList.style';
import { ListCard } from '../ListCard/ListCard';
const END_REACHED_TRESHOLD = 0.5;
export const IssuesList = ({ issues, handleOnEndReached, currentPageNumber }) => {

    const onEndReachedHandler = () => handleOnEndReached(currentPageNumber + 1);
    return (
        < FlatList style={styles.listStyle}
            data={issues}
            onEndReached={onEndReachedHandler}
            onEndReachedThreshold={END_REACHED_TRESHOLD}
            renderItem={({ item }) =>
                <ListCard
                    itemId={item.id}
                    itemTitle={item.title}
                />
            }
        />
    )
};

