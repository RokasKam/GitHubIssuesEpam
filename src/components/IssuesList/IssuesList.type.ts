export type Props = {
    issues: any;
    handleOnEndReached: (value: number) => void;
    currentPageNumber: number;
};