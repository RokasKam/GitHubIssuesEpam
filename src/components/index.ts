export { CheckBoxBlock } from "./CheckBoxBlock";
export { DropDownBlock } from "./DropDownBlock";
export { IssuesList } from "./IssuesList";
export { ListCard } from "./ListCard";
export { SearchButton } from "./SearchButton";
export { SearchInput } from "./SearchInput";

