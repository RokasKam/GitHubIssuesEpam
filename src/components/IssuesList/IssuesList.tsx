import React from 'react'
import { FlatList } from 'react-native';
import { styles } from './IssuesList.style';
import { ListCard } from '../ListCard';
import { Props } from './IssuesList.type';
const END_REACHED_TRESHOLD = 0.5;

export const IssuesList: React.FC<Props> = ({ issues, handleOnEndReached, currentPageNumber }) => {

    const onEndReachedHandler = () => handleOnEndReached(currentPageNumber + 1);
    return (
        < FlatList style={styles.listStyle}
            data={issues}
            onEndReached={onEndReachedHandler}
            onEndReachedThreshold={END_REACHED_TRESHOLD}
            renderItem={({ item }) =>
                <ListCard
                    itemId={item.id}
                    itemTitle={item.title}
                />
            }
        />
    )
};

