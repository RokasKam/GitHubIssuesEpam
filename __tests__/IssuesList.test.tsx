import React, { SetStateAction } from 'react';
import renderer from 'react-test-renderer';
import { IssuesList } from '../src/components/IssuesList'

test('renders correctly', () => {
  const tree = renderer.create(<IssuesList issues={undefined} handleOnEndReached={function (value: number): void {
    throw new Error('Function not implemented.');
  }} currentPageNumber={1} />).toJSON();
  expect(tree).toMatchSnapshot();
}); 