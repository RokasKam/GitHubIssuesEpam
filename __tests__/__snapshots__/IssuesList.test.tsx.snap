// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.5}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 30,
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View />
</RCTScrollView>
`;
